### ISSUE TITLE
Game leaks memory before reaching the login prompt on the Linux client

### ISSUE THREAD

**Issue by braiam** [StrangeLoopGames/EcoIssues#6815]
Steps to reproduce:

- Download eco 64-bit **client** for linux from SLG websites version 0.7.3.1
- Unzip Eco.x86_64 and Eco_Data directories
- Set Eco.x86_64 executable bit, using chmod +x
- Start the game
- Wait for the heat death of the universe or the oomkiller

According to zEverdeen on the #linux-server-discussion on discord, it "Looks like something didn't get compiled.". I couldn't verify if this is a regression since older clients doesn't seem to be available.

Player.log attached.

[Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/1865922/Player.log)

Mono 4.6.2.7 is also installed on the system since it's dependency to unrelated software (KeePass2), on a Debian testing system using xkde,

**Comment by braiam** [comment]
Same version on Windows 10 doesn't have the issue, same system.

**Comment by landru27** [comment]
I have this same exact symptom.

After launching the client, 'top' shows Eco pegged at 100% CPU (100% of just one CPU -- I have more than one core), and consuming more and more memory.  Eventually it wants more memory than physical RAM, so my system starts swapping, shifting the CPU load over to i/o wait, which just makes the problem worse.  And, Eco continues grabbing more and more memory.

Not willing to wait for oom-killer (nor the heat death of the universe), I HUP the process, it closes down, and my system returns to normal.

Ubuntu, 16.04.1, kernel 4.13.0-38
4 CPU i5 2.60GHz, 8 GB RAM
Eco v7.3.2 (I think; purchased on 4/5, but being unable to get past the splashscreen, I cannot see any game info directly)

**Comment by SeanAtr0n** [comment]
i'm guessing it was freezing up during shader warmup, that will be disabled for the next hotfix to help debug issues like this.

**Comment by braiam** [comment]
Please, make it a configurable parameter, so user can enable/disable it at will (be it a command line parameter, a ingame configuration or editing the configuration file).

**Comment by landru27** [comment]
I second @braiam 's comment about making it configurable;  I would just add that it would ideally be configurable from outside the running game as a base, and also in-game as an extra.  This, of course, because of the symptom of not being able to launch the game past the splash screen.

So, I would rephrase it as:  "a command line parameter or by editing the config file, and possibly also in-game".

Thanks!

**Comment by SeanAtr0n** [comment]
Should be resolved now, let me know if you still have issues
